src/metabase/api/action.clj

```diff
@@ -50,15 +52,32 @@
    [:parameter_mappings {:optional true} [:maybe map?]]])
 
 (api/defendpoint GET "/"
-  "Returns cards that can be used for QueryActions"
+  "Returns actions that can be used for QueryActions. By default lists all available actions. Pass optional
+  `?model-id=<model-id>` to limit to actions on a particular model."
   [model-id]
-  {model-id pos-int?}
-  (let [model (api/read-check Card model-id)]
-    ;; We don't check the permissions on the actions, we assume they are
-    ;; readable if the model is readable.
-    (hydrate
-     (action/select-actions [model] :model_id model-id :archived false)
-     :creator)))
+  {model-id [:maybe pos-int?]}
```

Comment:
using `ms/IntGreaterThanZero` instead of `pos-int?` will give a nicer error message.

Reply:
good thinking.